### PR TITLE
Add metainfo and install it

### PR DESCRIPTION
**Pull request by razzeee**
## Description

Provide a metainfo for appstream. So that we can offer distros to just copy this and Kodi will show up nicely in their software stores.

## Motivation and context

Mostly driven by the GNOME initative for the new GNOME Software

## How has this been tested?

Hasn't been tested. I'm also not sure if we want/need to implement it for the .deb build.

## What is the effect on users?

Nicer display in their software center.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by razzeee** [comment]
I think this file could also be translated https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-description but I'm not too sure.

@gade01

**Comment by gade01** [comment]
@razzeee 
Do you mean this file: https://github.com/xbmc/xbmc/blob/c820ef7adaf88aac7f5199a6620ff7e1691f60dd/tools/Linux/kodi.metainfo.xml.in

We absolutely could translate it.
Should translations go into separate folders or how should we do it?

**Comment by razzeee** [comment]
I think it needs to be done like this https://github.com/flathub/com.valvesoftware.Steam/blob/beta/com.valvesoftware.Steam.metainfo.xml#L17

**Comment by gade01** [comment]
Nice, that makes sense.
I just have to fiund out exactly what Weblate calls this file format.

**Comment by basilgello** [comment]
Good job @razzeee ! Once it is merged and reaches 19.2, I will include it to Debian packaging as we discussed on the forums.

> Do you mean this file: https://github.com/xbmc/xbmc/blob/c820ef7adaf88aac7f5199a6620ff7e1691f60dd/tools/Linux/kodi.metainfo.xml.in

> I think it needs to be done like this

Or is the metainfo.xml already built ? What needs to be improved in it then?

File: tools/Linux/kodi.metainfo.xml.in
Comment (garbear) [inline tools/Linux/kodi.metainfo.xml.in:32]: Great description! I would lead with the sentence concerning the "what":

> Kodi allows users to play and view most videos, music, podcasts, and other digital media files from local and network storage media and the internet.

File: cmake/scripts/linux/Install.cmake
Comment (wsnipex) [inline cmake/scripts/linux/Install.cmake:101]: absolute path please

Comment (razzeee) [reply cmake/scripts/linux/Install.cmake]: That actually was the wrong path too. Should be better now.

**Comment by basilgello** [comment]
LGTM. After this PR gets merged into Matrix, I will add `usr/share/metainfo` to `debian/kodi.install` list of Kodi packaging in Debian. @wsnipex you will likely need doing the same for Ubuntu PPA.

**Comment by razzeee** [comment]
I'll also update my flatpak PR when this lands. https://github.com/flathub/tv.kodi.Kodi/pull/115

File: tools/Linux/kodi.metainfo.xml.in
Comment (garbear) [reply tools/Linux/kodi.metainfo.xml.in:26]: Looks great

File: cmake/scripts/linux/Install.cmake
Comment (wsnipex) [reply cmake/scripts/linux/Install.cmake]: sorry, I guess I wasn't clear. We need to use 
${datarootdir}/metainfo
or it will break installing to alternate paths

**Review by garbear — APPROVED** [review]
LGTM once the absolute path is fixed to allow installing to custom locations

File: cmake/scripts/linux/Install.cmake
Comment (razzeee) [reply cmake/scripts/linux/Install.cmake]: Should be fixed now. I have no idea if that get's us to the correct folder, but we'll see :)

**Review by wsnipex — APPROVED** [review]

**Comment by wsnipex** [comment]
jenkins build this please

**Comment by wsnipex** [comment]
build failures look very much unlrelated. Closing/Reopening to get jenkins to update git rev

**Comment by razzeee** [comment]
Should this be backported?

**Comment by basilgello** [comment]
> Should this be backported?

Yes, please! If it reaches Matrix, I can add it to Debian packaging as I wrote :)

**Comment by razzeee** [comment]
https://github.com/flathub/tv.kodi.Kodi/pull/115#discussion_r720865039 pointed out, that there were multiple default entries. And while the spec allows for this, I don't think it makes any sense. So I removed all but one.

**Review by garbear — APPROVED** [review]

**Review by basilgello — APPROVED** [review]
LGTM. Pity this did not make its way into 19.2 :( I will go ahead and add this change as a Debian patch and push the backport

**Comment by razzeee** [comment]
I guess `tv.kodi.Kodi` is flathub specific then and other distros will probably have `kodi`. Not sure what's the best way to provide this for all consumers.

**Comment by razzeee** [comment]
I've also added the 19.2 release

**Comment by basilgello** [comment]
I went with `@APP_NAME_LC@` because it expands to `kodi` while `LC_APPNAME` is `Kodi`:

```diff
diff --git a/debian/patches/kodi/0022-Add-metainfo.patch b/debian/patches/kodi/0022-Add-metainfo.patch
index ba6c793c..22013a83 100644
--- a/debian/patches/kodi/0022-Add-metainfo.patch
+++ b/debian/patches/kodi/0022-Add-metainfo.patch
@@ -45,7 +45,7 @@ index 0000000000..ad273c3095
 @@ -0,0 +1,108 @@
 +<?xml version="1.0" encoding="UTF-8"?>
 +<component type="desktop-application">
-+    <id>tv.kodi.Kodi</id>
++    <id>@APP_NAME_LC@</id>
 +    <name>Kodi</name>
 +    <project_license>GPL-2.0-only GPL-2.0-or-later LGPL-2.1-or-later MIT BSD-3-Clause BSD-4-Clause</project_license>
 +    <metadata_license>CC0-1.0</metadata_license>
@@ -69,7 +69,7 @@ index 0000000000..ad273c3095
 +        very customisable and runs on a wide variety of devices. It is
 +        supported by a dedicated team of volunteers and a huge community.</p>
 +    </description>
-+    <launchable type="desktop-id">tv.kodi.Kodi.desktop</launchable>
++    <launchable type="desktop-id">@APP_NAME_LC@.desktop</launchable>
 +    <screenshots>
 +        <screenshot type="default">
 +            <image>https://mirrors.kodi.tv/screenshots/kodi-recently-added.jpg</image>
@@ -109,6 +109,7 @@ index 0000000000..ad273c3095
 +        </screenshot>
 +    </screenshots>
 +    <releases>
++        <release date="2021-10-08" version="19.2-Matrix"/>
 +        <release date="2021-05-09" version="19.1-Matrix"/>
 +        <release date="2021-02-19" version="19.0-Matrix"/>
 +        <release date="2020-10-24" version="18.9-Leia"/>
```

But yes, maybe first occurrence of `APP_NAME` is fine since we need a capitalized name in software catalogue.

**Comment by basilgello** [comment]
UPD: Looks like you basically repeated my patch :))) Cool!

**Comment by razzeee** [comment]
Updated

**Comment by razzeee** [comment]
I might need help with the flatpak after that

**Comment by razzeee** [comment]
> And two fixes in CMakeLists? :)

Sorry?

**Comment by basilgello** [comment]
@razzeee Finally Appstream gave me "validation successful" :) I have also updated the Matrix backport so you can merge both together.

**Pull request merged**
